src/pages/index.js

```diff
@@ -80,6 +56,165 @@ const cardLinkInput = addCardFormElement.querySelector(
 );
 const cardSelector = "#card-template";
 
+// Likes Api
+
+function handleCardLikeClick(cardId, isLiked) {
+  if (isLiked) {
+    api
+      .unlikeCard(cardId)
+      .then((cardData) => {
+        this._likes = cardData.isLiked;
+        this._renderLikes();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .likeCard(cardId)
+      .then((cardData) => {
+        this._likes = cardData.isLiked;
+        this._renderLikes();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+// Edit Avatar
+
+const editAvatarPopup = new PopupWithForm(
+  "#avatar-edit-modal",
+  handleEditAvatar,
+  "Save"
+);
+
+function handleEditAvatar(data) {
+  editAvatarPopup.setLoading(true);
+  api
+    .updateProfileAvatar(data.link)
+    .then((userData) => {
+      userInfo.setUserAvatar(userData.avatar);
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editAvatarPopup.setLoading(false, "Saving");
+    });
+}
+
+editAvatarPopup.setEventListeners();
+
+profileAvatarButton.addEventListener("click", () => editAvatarPopup.open());
+
+// User Info + Update Profile
+
+const userInfo = new UserInfo(profileTitle, profileDescription, avatarElement);
+
+function fillProfileForm() {
+  const userInfoData = userInfo.getUserInfo();
+  profileTitleInput.value = userInfoData.name;
+  profileDescriptionInput.value = userInfoData.description;
+}
+
+api.getUserInfo().then((userData) => {
+  userInfo.setUserInfo({
+    name: userData.name,
+    description: userData.about,
+  });
+});
+
+function handleProfileEditSubmit(data) {
+  editProfilePopup.setLoading(true);
+  api
+    .updateProfile(data)
+    .then((userData) => {
+      userInfo.setUserInfo({
+        name: userData.name,
+        description: userData.about,
+      });
+      fillProfileForm();
```

Comment:
`fillProfileForm()` is needed only when you open the profile popup. No need to call it everywhere

Reply:
* You haven't corrected something https://github.com/Gerson773/se_project_aroundtheus/pull/1#discussion_r1361337590